src/cel3ds/3dsmodel.cpp

```diff
@@ -21,230 +21,274 @@ M3DColor::M3DColor(float _red, float _green, float _blue) :
+void
+M3DMaterial::setTextureMap(const std::string& _texmap)
 {
     texmap = _texmap;
 }
 
 
-Eigen::Matrix4f M3DTriangleMesh::getMatrix() const
+Eigen::Matrix4f
+M3DTriangleMesh::getMatrix() const
 {
     return matrix;
 }
 
-void M3DTriangleMesh::setMatrix(const Eigen::Matrix4f& m)
+void
+M3DTriangleMesh::setMatrix(const Eigen::Matrix4f& m)
 {
     matrix = m;
 }
 
-Eigen::Vector3f M3DTriangleMesh::getVertex(std::uint16_t n) const
+Eigen::Vector3f
+M3DTriangleMesh::getVertex(std::uint16_t n) const
 {
     return points[n];
 }
 
-std::uint16_t M3DTriangleMesh::getVertexCount() const
+std::uint16_t
+M3DTriangleMesh::getVertexCount() const
 {
     return static_cast<std::uint16_t>(points.size());
 }
 
-void M3DTriangleMesh::addVertex(const Eigen::Vector3f& p)
+void
+M3DTriangleMesh::addVertex(const Eigen::Vector3f& p)
 {
     points.push_back(p);
 }
 
-Eigen::Vector2f M3DTriangleMesh::getTexCoord(std::uint16_t n) const
+Eigen::Vector2f
+M3DTriangleMesh::getTexCoord(std::uint16_t n) const
 {
     return texCoords[n];
 }
 
-std::uint16_t M3DTriangleMesh::getTexCoordCount() const
+std::uint16_t
+M3DTriangleMesh::getTexCoordCount() const
 {
     return static_cast<std::uint16_t>(texCoords.size());
 }
 
-void M3DTriangleMesh::addTexCoord(const Eigen::Vector2f& p)
+void
+M3DTriangleMesh::addTexCoord(const Eigen::Vector2f& p)
 {
     texCoords.push_back(p);
 }
 
-void M3DTriangleMesh::getFace(std::uint16_t n, std::uint16_t& v0, std::uint16_t& v1, std::uint16_t& v2) const
+void
+M3DTriangleMesh::getFace(std::uint16_t n, std::uint16_t& v0, std::uint16_t& v1, std::uint16_t& v2) const
 {
     int m = static_cast<int>(n) * 3;
     v0 = faces[m];
     v1 = faces[m + 1];
     v2 = faces[m + 2];
 }
 
-std::uint16_t M3DTriangleMesh::getFaceCount() const
+std::uint16_t
+M3DTriangleMesh::getFaceCount() const
 {
     return static_cast<std::uint16_t>(faces.size() / 3);
 }
 
-void M3DTriangleMesh::addFace(std::uint16_t v0, std::uint16_t v1, std::uint16_t v2)
+void
+M3DTriangleMesh::addFace(std::uint16_t v0, std::uint16_t v1, std::uint16_t v2)
 {
     faces.push_back(v0);
     faces.push_back(v1);
     faces.push_back(v2);
 }
 
-std::uint32_t M3DTriangleMesh::getSmoothingGroups(std::uint16_t face) const
+std::uint32_t
+M3DTriangleMesh::getSmoothingGroups(std::uint16_t face) const
 {
     return face < smoothingGroups.size() ? smoothingGroups[face] : 0;
 }
 
-void M3DTriangleMesh::addSmoothingGroups(std::uint32_t smGroups)
+void
+M3DTriangleMesh::addSmoothingGroups(std::uint32_t smGroups)
 {
     smoothingGroups.push_back(smGroups);
 }
 
-std::uint16_t M3DTriangleMesh::getSmoothingGroupCount() const
+std::uint16_t
+M3DTriangleMesh::getSmoothingGroupCount() const
 {
     return static_cast<std::uint16_t>(smoothingGroups.size());
 }
 
-void M3DTriangleMesh::addMeshMaterialGroup(std::unique_ptr<M3DMeshMaterialGroup> matGroup)
+void
+M3DTriangleMesh::addMeshMaterialGroup(M3DMeshMaterialGroup&& matGroup)
 {
     meshMaterialGroups.push_back(std::move(matGroup));
```

Comment:
could you update this with `emplace_back(matGroup);`?

Reply:
That would lead to this call doing a copy rather than a move (verified using a class that prints which constructor/assignment operator is called). Options that keep this as a move are `push_back(std::move(matGroup))` and `emplace_back(std::move(matGroup))`, and the `push_back` version feels more natural to me here.